src/main/java/me/gnat008/perworldinventory/data/metadata/PWIMetaDataManager.java

```diff
@@ -0,0 +1,37 @@
+package me.gnat008.perworldinventory.data.metadata;
+
+import ch.jalu.injector.Injector;
+
+import me.gnat008.perworldinventory.PerWorldInventory;
+
+import org.bukkit.metadata.MetadataValue;
+import org.bukkit.entity.Player;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class PWIMetaDataManager {
+
+    @Inject PerWorldInventory plugin;
+    @Inject Injector injector;
+
+    PWIMetaDataManager() {}
+    public <T extends PWIMetaDataValueAbstract> T createMetadataValue(Class<T> clazz, Object... args) {
```

Comment:
This is a lot of magic that can be simplified. You only have three (of which one is unused) implementations of `PWIMetaDataValueAbstract`. I'd suggest to have individual methods for creating these three classes, and to call the constructor manually.

```java
public PWIPlayerLastWorldInGroupData createLastWorldInGroupDataValue(Player player) {
   return new PWIPlayerLastWorldInGroupData(plugin, dataWriter, player);
}
```

i.e., 
1. Add `Player` to constructor of all PWIMetaDataValue classes where necessary
1. Remove `@Inject` from these constructors
1. Remove `init()` method; 
1. create individual methods in PWIMetaDataManager for creating metadata value classes